src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -200,48 +201,69 @@ public static GATKRead createReadAlignedToRef(final GATKRead originalRead,
         return Arrays.copyOfRange(bases, basesStart, basesStop + 1);
     }
 
-    public static byte[] getBasesAlignedOneToOne(final GATKRead read) {
-        return getSequenceAlignedOneToOne(read, r -> r.getBasesNoCopy(), GAP_CHARACTER);
-    }
-
-    public static byte[] getBaseQualsAlignedOneToOne(final GATKRead read) {
-        return getSequenceAlignedOneToOne(read, r -> r.getBaseQualitiesNoCopy(), (byte)0);
+    /**
+     * Returns the "IGV View" of all the bases and base qualities in a read aligned to the reference according to the cigar, dropping any bases
+     * that might be in the read but don't aren't in the reference. Any bases that appear in the reference but not the read
+     * will be filled in with GAP_CHARACTER values for the read bases and 0's for base qualities to indicate that they don't exist.
+     *
+     * If the cigar for input read is all matches to the reference then this method will return references to the original
+     * read base/base quality byte arrays in the underlying SamRecord in order to save on array allocation/copying performance effects.
+     *
+     * @param read a read to return aligned to the reference
+     * @return A tuple of byte arrays where the first array corresponds to the bases aligned to the reference and second
+     *         array corresponds to the baseQualities aligned to the reference.
+     */
+    public static Pair<byte[], byte[]> getBasesAndBaseQualitiesAlignedOneToOne(final GATKRead read) {
+        return getBasesAndBaseQualitiesAlignedOneToOne(read, GAP_CHARACTER, (byte)0);
     }
 
-    public static byte[] getSequenceAlignedOneToOne(final GATKRead read, final Function<GATKRead, byte[]> bytesProvider, final byte padWith) {
+    private static Pair<byte[], byte[]> getBasesAndBaseQualitiesAlignedOneToOne(final GATKRead read, final byte gapCharacter, final byte qualityPadCharacter) {
         Utils.nonNull(read);
-        Utils.nonNull(bytesProvider);
-        final Cigar cigar = read.getCigar();
-        final byte[] sequence = bytesProvider.apply(read);
-
-        if (!cigar.containsOperator(CigarOperator.DELETION) && !cigar.containsOperator(CigarOperator.INSERTION)) {
-            return sequence;
+        final byte[] bases = read.getBasesNoCopy();
+        final byte[] baseQualities = read.getBaseQualitiesNoCopy();
+        final int numCigarElements = read.numCigarElements();
+        boolean sawIndel = false;
+
+        // Check if the cigar contains indels
+        for (int i = 0; i < numCigarElements; i++) {
```

Comment:
Make a note explaining why you're not just calling `containsOperator()` twice here (otherwise someone may come along in a few years and change it back!)